data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchClientRefresher.java

```diff
@@ -54,8 +54,12 @@ public OpenSearchClient get() {
     }
 
     @Override
-    public void update(PluginSetting pluginSetting) {
-        final ConnectionConfiguration newConfig = ConnectionConfiguration.readConnectionConfiguration(pluginSetting);
+    public void update(final PluginSetting pluginSetting) {
```

Comment:
Since the class is implementing pluginComponentRefresher the method needs to be defined I think.

Reply:
ah okay, I would clarify in the comment a bit more if that's the case but no worries.

Reply:
I have implemented update function with OpenSearchSinkConfig below, i just needed to define a function that would take in PluginSettings or else I would get an error because of inheritance I think

Reply:
Change line 16 to this:

```
public class OpenSearchClientRefresher implements PluginComponentRefresher<OpenSearchClient, OpenSearchSinkConfig>
```